firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/settings/SettingsController.java

```diff
@@ -220,40 +248,52 @@ public Task<Void> then(@Nullable Void aVoid) throws Exception {
             });
   }
 
-  private Settings getCachedSettingsData(SettingsCacheBehavior cacheBehavior) {
-    Settings toReturn = null;
-
-    try {
-      if (!SettingsCacheBehavior.SKIP_CACHE_LOOKUP.equals(cacheBehavior)) {
-        final JSONObject settingsJson = cachedSettingsIo.readCachedSettings();
+  private Task<Settings> getCachedSettingsData(
```

Comment:
moves the read of the json settings into a Task as the caller can chain this call with another Task